### PR TITLE
[admin] Add CheckboxLists in role form

### PR DESCRIPTION
**Pull request by arnaud-morvan**
Note that this also refactor requirements between commons and admin to avoid having two different references to c2cgeoform.

### CODE REVIEW

File: docker/build/requirements.txt [hunk 1]
```diff
@@ -5,9 +5,9 @@ boto==2.48.0  # Tile generation
 bottle==0.12.13  # i18n
 c2c.cssmin==0.7.dev6  # CGXP build
 c2c.template==1.5.1  # geoportal
-https://github.com/camptocamp/c2cgeoform/archive/87a4191ef3330b76497bd009e9c0220cbc73c625.zip#egg=c2cgeoform  # commons
+-e git://github.com/camptocamp/c2cgeoform.git@b6d4bbc78da55bf270ae62ee3eaac77d5dc92062#egg=c2cgeoform  # commons
```
Comment (sbrunner) [inline docker/build/requirements.txt]: Don't use the git protocol, it's working only it we have an ssh key, ant he archive way download less data ...

Comment (jwkaltz) [reply docker/build/requirements.txt]: But if we use the git protocol, this will work behind proxies (whereas https: will create problems), IIUC?

**Comment by arnaud-morvan** [comment]
@sbrunner: I do not understand what happend. Model main perfectly works in admin.
I've tried to run one test locally with:
```
./docker-compose-run py.test -k test_get_class_nonexisting_table --cov=geoportal/c2cgeoportal_geoportal geoportal/tests
```
But seems that setup_common is not executed.
I do not understand `setup_method`, for me we use fixtures with pytests.

File: docker/build/requirements.txt (hunk 1 above)
Comment (arnaud-morvan) [reply docker/build/requirements.txt]: @jwkaltz: I do not understand your comments, proxy usually allow https but not git, in aeroportdelyon_sig we are behin a proxy and we use: 
```
git config --global url."https://".insteadOf git://
```
So this is transparent except you have to give your credentials for private repositories.

File: docker/build/requirements.txt [hunk 2]
```diff
@@ -5,9 +5,9 @@ boto==2.48.0  # Tile generation
 bottle==0.12.13  # i18n
 c2c.cssmin==0.7.dev6  # CGXP build
 c2c.template==1.5.1  # geoportal
-https://github.com/camptocamp/c2cgeoform/archive/87a4191ef3330b76497bd009e9c0220cbc73c625.zip#egg=c2cgeoform  # commons
+https://github.com/camptocamp/c2cgeoform/archive/b6d4bbc78da55bf270ae62ee3eaac77d5dc92062.zip  # commons
```
Comment (sbrunner) [inline docker/build/requirements.txt]: I think that we should keep the egg=

File: docker/build/requirements.txt [hunk 3]
```diff
@@ -5,9 +5,9 @@ boto==2.48.0  # Tile generation
...
 codespell==1.10.0  # Lint
-ColanderAlchemy==0.3.3  # commons
+https://github.com/arnaud-morvan/ColanderAlchemy/archive/f61482e426a6ca97c31a254647aa77f5419f31f3.zip  # commons
```
Comment (sbrunner) [inline docker/build/requirements.txt]: Also ass ?egg=ColanderAlchemy :-)

**Comment by sbrunner** [comment]
I will test it tomorrow ...

File: docker/build/requirements.txt (hunk 1 above)
Comment (jwkaltz) [reply docker/build/requirements.txt]: @arnaud-morvan I think I mixed things up. I was thinking about https://github.com/camptocamp/ngeo/pull/2876

**Comment by arnaud-morvan** [comment]
Concerning travis error, I've put in docker/build/requirements.txt:
```
https://github.com/camptocamp/c2cgeoform/archive/d61503840067fae7e956783924d831b7403490b1.zip#egg=c2cgeoform  # commons
codespell==1.10.0  # Lint
https://github.com/arnaud-morvan/ColanderAlchemy/archive/f61482e426a6ca97c31a254647aa77f5419f31f3.zip#egg=ColanderAlchemy  # commons
```
but pip freeze gives :

**Comment by arnaud-morvan** [comment]
```
-e git://github.com/camptocamp/c2cgeoform.git@b6d4bbc78da55bf270ae62ee3eaac77d5dc92062#egg=c2cgeoform
-e git+git@github.com:camptocamp/c2cgeoportal.git@efed9b81e95761ae5a4e52d9c1201895fdb13839#egg=c2cgeoportal_admin&subdirectory=admin
-e git+git@github.com:camptocamp/c2cgeoportal.git@efed9b81e95761ae5a4e52d9c1201895fdb13839#egg=c2cgeoportal_commons&subdirectory=commons
-e git+git@github.com:camptocamp/c2cgeoportal.git@efed9b81e95761ae5a4e52d9c1201895fdb13839#egg=c2cgeoportal_geoportal&subdirectory=geoportal
ColanderAlchemy==0.3.4.dev1
```
I do not understand why c2cgeoportal packages are listed here, I do not found where they are installed.

**Comment by arnaud-morvan** [comment]
And I do not understand why colanderalchemy is not freezed as github url

**Comment by arnaud-morvan** [comment]
I've proposed a PR in c2cgeoform : https://github.com/camptocamp/c2cgeoform/pull/134
Next week I will come back from ColanderAlchemy patching in GMF Admin, and from model modifications. So problems above may be solved.

**Comment by arnaud-morvan** [comment]
@sbrunner: waiting for CI, but this may ready for review.
Note that I've made minimum of modifications to main model to avoid problems with geoportal tests.

File: commons/c2cgeoportal_commons/models/main.py [hunk 4]
```diff
@@ -160,6 +163,16 @@ def __unicode__(self) -> str:
 )
 
 
+@colander.deferred
+def defered_restrictionareas_widget(node: colander.SchemaNode, kw: dict) -> Widget:
+    # pylint: disable=unused-argument
```
[the remark this answers is not in the document]
Comment (arnaud-morvan) [reply commons/c2cgeoportal_commons/models/main.py]: I've removed this function that is not used anymore since we moved ManyMany to views intead of model.

**Pull request merged**